feat: Add categories with icons

lukasdotcom opened the pull request:
Resolves: #299
Resolves: #300
Resolves: #301
<img width="1912" height="1237" alt="image" src="https://github.com/user-attachments/assets/a9bc031c-f9ea-45b6-a4fb-2225d439537a" />

### Some design questions.

1. For some reason the waveform icon is different in the vue material icons than on the material icons (or I couldn't find it🤷) so had to upload the icon from google fonts. The one that vue material icons comes with is: <img width="827" height="191" alt="image" src="https://github.com/user-attachments/assets/5aa02177-a878-4918-83ba-debb6986bf74" />

2. I'm not sure if context chat should have its own category, but left it in text (can be moved to other). EDIT: Added category for context chat with icon that was discussed with @marcoambrosini 

3. I changed the maximum width to 1220px instead of 1200px to ensure that all categories can fit on one line in english.

### Other notes.
1. The rules to assign categories are built to be flexible and guess categories for new tasks as long as they follow some implied rules. 
2. I did remove the inline menu, but theoretically could add the old one back just doesn't work well with categories.
3. Also added convenience feature that if only one task exists in category clicking on the category will bring you to that task.

marcelklehr commented:
I'd be in favor of moving context chat out of the Working with text category

lukasdotcom commented:
The last commit added an overflow menu that allows you to select categories and tasks. Note that the amount of inlined tasks needs to be specified as a prop currently by the parent.
<img width="1912" height="1237" alt="image" src="https://github.com/user-attachments/assets/61a792e8-d18b-4788-aa1e-6080d1eda70b" />

julien-nc commented on src/components/TaskTypeSelect.vue:
```diff
@@ -78,66 +115,137 @@ export default {
-				? 'primary'
-				: 'secondary'
+		},
+		selectedCategory(category) {
```
```suggestion
		isCategorySelected(category) {
```
More explicit IMO

julien-nc commented on src/components/TaskTypeSelect.vue:
```diff
@@ -78,66 +115,137 @@ export default {
+			if (this.onlyInline) {
+				return { inline: this.buttonTypes, overflow: [] }
+			}
+			const inline = this.buttonTypes.slice(0, this.inline)
+			let overflow = this.buttonTypes.slice(this.inline)
```
```suggestion
			const inlineButtonTypes = this.buttonTypes.slice(0, this.inline)
			let overflowButtonTypes = this.buttonTypes.slice(this.inline)
```
Maybe?

julien-nc commented:
Also, I would recommend to rebase on main from time to time. A small conflict is better than a big conflict :grin:

lukasdotcom commented:
> A bit of nitpicking 😁 Also, could we prevent opening the actions dropdown when clicking on a category in which there is only one task type? Maybe we should only do that when the category name equals the single task type name. It's the case for translate and chat with AI.

I changed it to create a NcButton when there is no dropdown needed because I couldn't find a way to hide it and implemented the other minor name changes.

jancborchardt reviewed (commented):
Looks really good @lukasdotcom! :) It will help people a lot to have a better overview
